libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.spec.ts

```diff
@@ -170,18 +175,106 @@ describe('ChartJSService', () => {
+          const labels = ['one', 'two', 'three'];
+          expect(labels.length).toBeGreaterThan(0);
+          expect(data.length).toBeGreaterThan(0);
+
+          chartJSService.renderChart({
+            targetElement: canvasElement,
+            type: 'bar',
+            data,
+            labels,
+          });
+
+          const chartLabels = chartJSService['chart'].data.labels;
+          expect(chartLabels.length).toEqual(labels.length);
+          expect(chartLabels).toEqual(labels);
+        });
+
+        describe('and the dataset also contains labels', () => {
+          it('should apply argument labels', () => {
+            const data = [
+              {
+                data: [
+                  { x: 'en', y: 1 },
+                  { x: 'to', y: 2 },
+                  { x: 'tre', y: 3 },
+                ],
+              },
+            ] as any;
+            const labels = ['one', 'two', 'three'];
+            expect(labels.length).toBeGreaterThan(0);
+            expect(data.length).toBeGreaterThan(0);
+
+            chartJSService.renderChart({
+              targetElement: canvasElement,
+              type: 'column',
+              data,
+              labels,
+            });
+
+            const chartLabels = chartJSService['chart'].data.labels;
+            expect(chartLabels.length).toEqual(labels.length);
+            expect(chartLabels).toEqual(labels);
+          });
+        });
+      });
+    });
+
+    describe('when no labels are provided', () => {
       it('should have a blank label for each data point', () => {
+        const data = [1, 2, 3];
+        const expectedLabels = ['', '', ''];
+        expect(data.length).toBeGreaterThan(0);
+        expect(expectedLabels.length).toEqual(data.length);
+
         chartJSService.renderChart({
           targetElement: canvasElement,
           type: 'column',
-          data: [1, 2, 3],
+          data,
         });
 
         const chartLabels = chartJSService['chart'].data.labels;
-        expect(chartLabels.length).toEqual(3);
-        chartLabels.forEach((label) => {
-          expect(label).toEqual('');
+        expect(chartLabels.length).toEqual(data.length);
+        expect(chartLabels).toEqual(expectedLabels);
+      });
+
+      describe('but the dataset contains labels for the index axis', () => {
+        it('should use the dataset labels', () => {
+          const data = [
+            { x: 'en', y: 1 },
+            { x: 'to', y: 2 },
+            { x: 'tre', y: 3 },
+          ] as any;
+          const dataset = [
+            {
+              data,
+            },
+          ];
+          expect(data.length).toBeGreaterThan(0);
+          expect(data.every(({ x }) => typeof x === 'string')).toBeTrue();
+
+          chartJSService.renderChart({
+            targetElement: canvasElement,
+            type: 'column',
+            data: dataset,
+          });
+
+          /* This assertion relies on the assumption that:
```

Comment:
ice with an explanation for the less obvious stuff. Consider if the actual implementation also might benefit from an explanation regarding which labels are to be handled and how they are processed - assumtions etc. It's very difficult to read from the code alone and weak or missing types doesn't help either.